File: Movie/Movie.Web/Modules/ServerTypes/Club/ClubColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { ClubRow } from "./ClubRow";

export interface ClubColumns {
    ClubId: Column<ClubRow>;
    ClubName: Column<ClubRow>;
    Code: Column<ClubRow>;
    PingTime: Column<ClubRow>;
}

export class ClubColumns extends ColumnsBase<ClubRow> {
    static readonly columnsKey = 'Club.Club';
    static readonly Fields = fieldsProxy<ClubColumns>();
}

File: Movie/Movie.Web/Modules/Club/Club/ClubColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Movie.Club.Columns;

[ColumnsScript("Club.Club")]
[BasedOnRow(typeof(ClubRow), CheckNames = true)]
public class ClubColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ClubId { get; set; }
    [EditLink]
    public string ClubName { get; set; }
    public string Code { get; set; }
    public DateTime PingTime { get; set; }
}